src/common/com/intellij/plugins/haxe/ide/index/HaxeComponentFileNameIndex.java

```diff
@@ -99,17 +101,26 @@ public static boolean getFileNames(@NotNull String qName,
     @NotNull
     public Map<String, Void> map(final FileContent inputData) {
       final PsiFile psiFile = inputData.getPsiFile();
+      final String moduleName = FileUtil.getNameWithoutExtension(inputData.getFileName());
       final List<HaxeClass> classes = HaxeResolveUtil.findComponentDeclarations(psiFile);
       if (classes.isEmpty()) {
         return Collections.emptyMap();
```

Comment:
If I got right your idea, then if we have file

`Main.hx` which doesn't contains `Main` class, then instead of ignoring such files (and not creating cache for them), we still create item in cache

Then everything else, seems like already implemented (by using something like `findClassByQName`)

If file doesn't contains classes, what else useful it can provide? I mean typedef is considered to be a HaxeClass too in this system, if I got it right